src/frontend/components/PublicPausedLiveVideo/index.tsx

```diff
@@ -0,0 +1,95 @@
+import { Box, Clock, Layer, Paragraph } from 'grommet';
+import { DateTime } from 'luxon';
+import React, { useEffect, useState } from 'react';
+import { defineMessages, FormattedMessage } from 'react-intl';
+import videojs from 'video.js';
+
+import { H2 } from '../Headings';
+import { Video } from '../../types/tracks';
+import { useAsyncEffect } from 'utils/useAsyncEffect';
+import { resumeLive } from '../../utils/resumeLive';
+
+const messages = defineMessages({
+  pausedSince: {
+    defaultMessage: 'The webinar is in paused since ',
+    description: 'Information about how many time the webinar is paused.',
+    id: 'components.PublicPausedLive.pausedSince',
+  },
+  title: {
+    defaultMessage: 'Webinar is paused',
+    description: 'Title for the public paused live component',
+    id: 'components.PublicPausedLive.title',
+  },
+  text: {
+    defaultMessage:
+      'The webinar is paused. When resumed, the video will start again.',
+    description: 'Message for the public paused live component',
+    id: 'components.PublicPausedLive.message',
+  },
+});
+
+interface PublicPausedLiveVideoProps {
+  video: Video;
+  videoNodeRef: HTMLVideoElement;
+}
+
+export const PublicPausedLiveVideo = ({
+  video,
+  videoNodeRef,
+}: PublicPausedLiveVideoProps) => {
+  const [time, setTime] = useState<string>();
+
+  useEffect(() => {
+    if (video.live_info?.paused_at) {
+      const now = DateTime.now();
+      const pausedAt = DateTime.fromSeconds(Number(video.live_info?.paused_at));
+      const diff = now
+        .diff(pausedAt, ['hours', 'minutes', 'seconds'])
+        .toObject();
+      // Using diff.toISO() would be better but Clock grommet component is not able to manage
+      // iso8601 period without hours.
+      setTime(`PT${diff.hours}H${diff.minutes}M${diff.seconds}S`);
+    }
+  }, []);
+
+  useAsyncEffect(async () => {
+    await resumeLive(video);
+    const player = Object.values(videojs.getPlayers())[0];
+    player.src(player.currentSource());
+  }, []);
+
+  return (
```

Comment:
in a general manner, I like to split component with business logic (like computing timestamp since when the live is paused) from component used in the layout process (like using a clock, etc)
this is for better reusability of graphical components and easier testing in business logic

Reply:
So here every `useMemo` callback should be in the dedicated module?